### PR TITLE
Add text-gen notebook for llama models

### PR DESCRIPTION
**Pull request by aggarwal-k**
# Description

# Checklist

- [ ] I have read the [contribution guidelines](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
- [ ] I have coordinated with the docs team (mldocs@microsoft.com) if this PR deletes files or changes any file names or file extensions.
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by jpmann** [comment]
fix the readme link for pipeline parameters

**Comment by jpmann** [comment]
create a separate data section for advanced users

File: sdk/python/foundation-models/system/finetune/Llama-notebooks/text-generation/download-dataset.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+args = parser.parse_args()
+
+# create the download directory if it does not exist
+if not os.path.exists(args.download_dir):
+    os.makedirs(args.download_dir)
+
+
+# import hugging face datasets library
+from datasets import load_dataset, get_dataset_split_names
+from functools import partial
```
Comment (jpmann) [inline sdk/python/foundation-models/system/finetune/Llama-notebooks/text-generation/download-dataset.py:27]: I don't see partial being used. Can we remove it?

File: sdk/python/foundation-models/system/finetune/Llama-notebooks/text-generation/text-generation-samsum.ipynb [hunk 2]
```diff
@@ -0,0 +1,794 @@
...
+    "\n",
+    "This sample shows how use `text-generation` components from the `azureml` system registry to fine tune a model to summarize a dialog between 2 people using samsum dataset. We then deploy the fine tuned model to an online endpoint for real time inference.\n",
+    "\n",
+    "### Training data\n",
+    "We will use the [samsum](https://huggingface.co/datasets/samsum) dataset. This dataset is intended to summarize dialogues between 2 people. with this notebook we will summarize the dialogues and calculate bleu and rouge scores for the summarized text vs provided ground_truth summaries\n",
+    "\n",
+    "### Model\n",
+    "We will use the `llama-2-7b` model to show how user can finetune a model for text-generation task. If you opened this notebook from a specific model card, remember to replace the specific model name. Optionally, if you need to fine tune a model that is available on HuggingFace, but not available in `azureml` system registry, you can either [import](https://github.com/Azure/azureml-examples) the model or use the `huggingface_id` parameter instruct the components to pull the model directly from HuggingFace. \n",
```
Comment (jpmann) [inline sdk/python/foundation-models/system/finetune/Llama-notebooks/text-generation/text-generation-samsum.ipynb:16]: Any idea why the import is pointed to azureml-examples?

Comment (jpmann) [reply sdk/python/foundation-models/system/finetune/Llama-notebooks/text-generation/text-generation-samsum.ipynb:16]: I see the same for another task as well. I think this is the right link - https://github.com/Azure/azureml-examples/blob/main/sdk/python/foundation-models/system/import/import_model_into_registry.ipynb

File: sdk/python/foundation-models/system/finetune/Llama-notebooks/text-generation/text-generation-samsum.ipynb [hunk 3]
```diff
@@ -0,0 +1,794 @@
...
+    "        f\"This should not happen. Please check the selected compute cluster: {compute_cluster} and try again.\"\n",
+    "    )"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 4. Pick the dataset for fine-tuning the model\n",
```
Comment (jpmann) [inline sdk/python/foundation-models/system/finetune/Llama-notebooks/text-generation/text-generation-samsum.ipynb:264]: Shall we add a new section regarding multi-node?

File: sdk/python/foundation-models/system/finetune/Llama-notebooks/text-generation/text-generation-samsum.ipynb [hunk 4]
```diff
@@ -0,0 +1,794 @@
...
+    "        ground_truth_key=\"summary\",\n",
+    "        # Training settings\n",
+    "        number_of_gpu_to_use_finetuning=gpus_per_node,  # set to the number of GPUs available in the compute\n",
+    "        **training_parameters,\n",
+    "        **optimization_parameters\n",
+    "    )\n",
+    "    return {\n",
+    "        # map the output of the fine tuning job to the output of pipeline job so that we can easily register the fine tuned model\n",
+    "        # registering the model is required to deploy the model to an online or batch endpoint\n",
+    "        \"trained_model\": text_classification_pipeline.outputs.mlflow_model_folder\n",
```
Comment (jpmann) [inline sdk/python/foundation-models/system/finetune/Llama-notebooks/text-generation/text-generation-samsum.ipynb:483]: typo  change text_classification_pipeline -> text_generation_pipeline

File: sdk/python/foundation-models/system/finetune/Llama-notebooks/text-generation/text-generation-samsum.ipynb [hunk 5]
```diff
@@ -0,0 +1,794 @@
...
+    ")\n",
+    "\n",
+    "\n",
+    "# define the pipeline job\n",
+    "@pipeline()\n",
+    "def create_pipeline():\n",
+    "    text_classification_pipeline = pipeline_component_func(\n",
+    "        # specify the foundation model available in the azureml system registry id identified in step #3\n",
+    "        mlflow_model_path=foundation_model.id,\n",
+    "        # huggingface_id = 'bert-base-uncased', # if you want to use a huggingface model, uncomment this line and comment the above line\n",
```
Comment (jpmann) [inline sdk/python/foundation-models/system/finetune/Llama-notebooks/text-generation/text-generation-samsum.ipynb:458]: shall we replace this _huggingface_id_ with llama's huggingface id?

File: sdk/python/foundation-models/system/finetune/Llama-notebooks/text-generation/text-generation-samsum.ipynb (hunk 3 above)
Comment (jpmann) [reply sdk/python/foundation-models/system/finetune/Llama-notebooks/text-generation/text-generation-samsum.ipynb:264]: will add as new notebook

File: sdk/python/foundation-models/system/finetune/Llama-notebooks/text-generation/text-generation-samsum.ipynb [hunk 6]
```diff
@@ -0,0 +1,794 @@
...
+    "# fetch the pipeline component\n",
+    "pipeline_component_func = registry_ml_client.components.get(\n",
+    "    name=\"text_generation_pipeline\", label=\"latest\"\n",
+    ")\n",
+    "\n",
+    "\n",
+    "# define the pipeline job\n",
+    "@pipeline()\n",
+    "def create_pipeline():\n",
+    "    text_classification_pipeline = pipeline_component_func(\n",
```
[the remark this answers is not in the document]
Comment (aggarwal-k) [reply sdk/python/foundation-models/system/finetune/Llama-notebooks/text-generation/text-generation-samsum.ipynb]: Edited

File: sdk/python/foundation-models/system/finetune/Llama-notebooks/text-generation/download-dataset.py (hunk 1 above)
Comment (aggarwal-k) [reply sdk/python/foundation-models/system/finetune/Llama-notebooks/text-generation/download-dataset.py]: Removed

File: sdk/python/foundation-models/system/finetune/Llama-notebooks/text-generation/text-generation-samsum.ipynb (hunk 2 above)
Comment (aggarwal-k) [reply sdk/python/foundation-models/system/finetune/Llama-notebooks/text-generation/text-generation-samsum.ipynb]: yeah, changed for text gen now, will pick up other tasks later

File: sdk/python/foundation-models/system/finetune/Llama-notebooks/text-generation/text-generation-samsum.ipynb (hunk 5 above)
Comment (aggarwal-k) [reply sdk/python/foundation-models/system/finetune/Llama-notebooks/text-generation/text-generation-samsum.ipynb]: Changed anyway

File: sdk/python/foundation-models/system/finetune/Llama-notebooks/text-generation/text-generation-samsum.ipynb (hunk 4 above)
Comment (aggarwal-k) [reply sdk/python/foundation-models/system/finetune/Llama-notebooks/text-generation/text-generation-samsum.ipynb]: Fixed

**Review by amangupta26 — APPROVED** [review]

**Review by jpmann — APPROVED** [review]

**Pull request merged**